### PR TITLE
check that bots are allowed to evolve/buy in ::allowed()

### PR DESCRIPTION
**Pull request by bmorel**
This complements a change in #2190 while not conflicting.
Notably, this removes the need for https://github.com/Unvanquished/Unvanquished/pull/2190/commits/e34f6cd8e9c4f5ffdf379237552e5cc53314f4b3#diff-7efa7a4eedeb7b61fdb165d5390858216282e89de0c7009041d7004243d50303R507 to still check for `g_bot_evolve.Get()` as it makes little sense that the sg_bot_util.cpp functions do not do it already.

cc @necessarily-equal

### CODE REVIEW

**Review by necessarily-equal — APPROVED** [review]
LGTM, haven't tested

**Review by DolceTriade — APPROVED** [review]

**Comment by bmorel** [comment]
Re-reading it, this code smells... at the very least I need to add comments to say that only aliens "buy classes".

File: src/sgame/sg_bot_util.cpp [hunk 1]
```diff
@@ -127,19 +127,19 @@ bool equipment_t<weapon_t>::unlocked( void ) const
 template <>
 bool equipment_t<class_t>::allowed( void ) const
 {
```
Comment (bmorel) [inline src/sgame/sg_bot_util.cpp:129]:
```suggestion
{
// HACK: only aliens can "buy" classes, aka evolve, so this should work, for now.
```

File: src/sgame/sg_bot_util.cpp [hunk 2]
```diff
@@ -127,19 +127,19 @@ bool equipment_t<weapon_t>::unlocked( void ) const
...
-	return authorized.Get() && !BG_ClassDisabled( item );
+	return authorized.Get() && g_bot_evolve.Get() && !BG_ClassDisabled( item );
 }
 
 template <>
 bool equipment_t<upgrade_t>::allowed( void ) const
 {
```
Comment (bmorel) [inline src/sgame/sg_bot_util.cpp:135]:
```suggestion
{
// HACK: only humans can buy equipment, so this should be fine for now.
```

File: src/sgame/sg_bot_util.cpp [hunk 3]
```diff
@@ -127,19 +127,19 @@ bool equipment_t<weapon_t>::unlocked( void ) const
...
-	return authorized.Get() && !BG_UpgradeDisabled( item );
+	return authorized.Get() && g_bot_buy.Get() && !BG_UpgradeDisabled( item );
 }
 
 template <>
 bool equipment_t<weapon_t>::allowed( void ) const
 {
```
Comment (bmorel) [inline src/sgame/sg_bot_util.cpp:141]:
```suggestion
{
// HACK: only humans can buy equipment, so this should be fine for now.
```

File: src/sgame/sg_bot_util.cpp [hunk 4]
```diff
@@ -127,19 +127,22 @@ bool equipment_t<weapon_t>::unlocked( void ) const
...
-	return authorized.Get() && !BG_UpgradeDisabled( item );
+// HACK: only humans can buy equipment, so this should be fine for now.
+	return authorized.Get() && g_bot_buy.Get() && !BG_UpgradeDisabled( item );
 }
 
 template <>
 bool equipment_t<weapon_t>::allowed( void ) const
 {
-	return authorized.Get() && !BG_WeaponDisabled( item );
+// HACK: only humans can buy equipment, so this should be fine for now.
```
Comment (slipher) [inline src/sgame/sg_bot_util.cpp:144]: I don't get it. What's team-specific about the code here?

Comment (bmorel) [reply src/sgame/sg_bot_util.cpp:144]: right now, only humans can buy

Comment (bmorel) [reply src/sgame/sg_bot_util.cpp:144]: I can remove comments if you want, but this code now makes me worries about future changes.

Comment (slipher) [reply src/sgame/sg_bot_util.cpp:144]: But nothing in this line of code mentions a specific team.

Comment (bmorel) [reply src/sgame/sg_bot_util.cpp:144]: I guess that's true.

**Pull request merged**